### ISSUE TITLE
I have encountered this problem and I couldn't overcome. Could you fix it if you did might you help about how to fix? I didn't understand other commend.

### ISSUE THREAD

**Issue by Anushkatech5** [pypa/packaging-problems#738]
I have encountered this problem and I couldn't overcome. Could you fix it if you did might you help about how to fix? I didn't understand other command.
              IN WINDOWS also, please solve the same issue.

![image](https://github.com/pypa/packaging-problems/assets/123286374/cdf82d5c-6b60-4489-8d98-18a509030ddf)

_Originally posted by @mmehmetaliyavuz in https://github.com/pypa/packaging-problems/issues/681#issuecomment-1645270111_

**Comment by merwok** [comment]
Have you tried installing microsoft visual c++ 14.0 or greater by following the link in the last line of the error message?

**Comment by Anushkatech5** [comment]
Yes, I've installed it correctly! But still not working. What should I do?

**Comment by jeanas** [comment]
@Anushkatech5 You are on Python 3.7. PyQt5 does not have precompiled packages for Python 3.7, and is a huge beast to build yourself. I strongly recommend using a newer Python version, there you will get a precompiled package which should just work.

**Comment by Anushkatech5** [comment]
Thanks for the tip!
Actually, Am using this python version just because I also need to use
tensorflow and tensorflow-gpu which are compatible with python 3.7. And
also can you please suggest which python version would be more suitable for
PyQt5?

**Comment by henryiii** [comment]
TensorFlow added 3.12 support a few weeks ago. But to be safe, 3.11 is probably fine? TensorFlow requires 3.9+ in its most recent versions, FYI.

**Comment by Anushkatech5** [comment]
Yes, my error is fixed and all are compatible with python 3.8. As of
tensorflow-gpu there will be some errors with python 3.9 or python 3.9+. So
best bet is python 3.8! Thanks for your help!

**Comment by henryiii** [comment]
`tensorflow-gpu` is abandoned since `tensorflow` is exactly the same package! From the PyPI page for tensorflow-gpu:

> tensorflow and tensorflow-gpu have been the same package since TensorFlow 2.1, released in September 2019. Although the checksums differ due to metadata, they were built in the same way and both provide GPU support via Nvidia CUDA. As of December 2022, tensorflow-gpu has been removed and has been replaced with this new, empty package that generates an error upon installation.
All existing versions of tensorflow-gpu are still available, but the TensorFlow team has stopped releasing any new tensorflow-gpu packages, and will not release any patches for existing tensorflow-gpu versions.
